src/algebra/graded_monoid.lean

```diff
@@ -394,3 +395,28 @@ instance set_like.gcomm_monoid {S : Type*} [set_like S R] [comm_monoid R] [add_c
   ..set_like.gmonoid A}
 
 end subobjects
+
+section homogeneous_elements
+
+variables {R S : Type*} [set_like S R]
+
+/-- An element `a : R` is said to be homogeneous if there is some `i : ι` such that `a ∈ A i`-/
+def set_like.is_homogeneous (A : ι → S) (a : R) : Prop := ∃ i, a ∈ A i
+
+lemma set_like.is_homogeneous_one [has_zero ι] [has_one R]
+  (A : ι → S) [set_like.has_graded_one A] : set_like.is_homogeneous A (1 : R) :=
+⟨0, set_like.has_graded_one.one_mem⟩
+
+lemma set_like.is_homogeneous.mul [has_add ι] [has_mul R] {A : ι → S}
```

Comment:
I guess we should add `is_homogenous_zero_submodule` and `is_homogenous.smul_submodule` to `direct_sum/internal` too